ingest-app/src/cmr/ingest/api/subscriptions.clj

```diff
@@ -57,6 +59,40 @@
            subscriber-id
            concept-id)))))
 
+(defn- subscription-duplicate-error
+  "Will Throw an error explaining that subscriptions can not be duplicated"
+  [subscriber-id concept-id, query]
+  (errors/throw-service-error
+   :conflict
+   (format (str "The subscriber id [%s] has already subscribed to the "
+                "collection with concept id [%s] using the query [%s]. "
+                "Subscribers must use unique queries for each Collection.")
+           subscriber-id
+           concept-id
+           query)))
+
+(defn- check-duplicate-subscription
+  "The query used by a subscriber for a collection should be unique to prevent
+   redundent emails from being sent to them. This function will check that a
+   subscription is unique for the following conditions: native-id, collection-id,
+   subscriber-id, normalized-query."
+  [request-context subscription]
+  (let [native-id (:native-id subscription)
+        provider-id (:provider-id subscription)
+        metadata (-> (:metadata subscription) (json/decode true))
+        normalized-query (:normalized-query subscription)
+        collection-id (:CollectionConceptId metadata)
+        subscriber-id (:SubscriberId metadata)]
+    ;; There should be no other subscriptions matching this query ; nil is good
+    (when-not (nil? (mdb/find-latest-concept
```

Comment:
change this to (when (mdb/find-latest-concept...). Also fix the indentations in the lines below.